### PR TITLE
Add cloudwatch monitoring

### PR DESCRIPTION
**Pull request by AndrewGorton**
If the config value "cloudwatch.environment.name" is set, the indexer will write a value of 1.0 to CloudWatch every 10 seconds using the config value as the namespace. You can [monitor this metric](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#metrics:graph=~%28metrics~%28~%28~'local~'heartbeat~'application~'indexer~%28yAxis~'right%29%29%29~period~60~stat~'SampleCount~start~'-PT4H~end~'P0D~yAxis~%28left~null~right~null%29~region~'eu-west-1%29) for ALARM or INSUFFICIENT to detect whether the indexer process is running.

AWS Credentials need passing using [one of the standard mechanisms](http://docs.aws.amazon.com/AWSSdkDocsJava/latest/DeveloperGuide/credentials.html) (eg environment variables or command line java -D<definition> -jar indexer.jar).

### CODE REVIEW

File: src/main/java/uk/gov/indexer/Application.java [hunk 1]
```diff
@@ -24,6 +26,11 @@ public static void main(String[] args) throws InterruptedException {
 
         indexers.stream().parallel().forEach(indexer -> indexer.start(executorService));
 
+        if (configuration.cloudwatchEnvironmentName().isPresent()) {
+            heartbeatThread = new Thread(new CloudWatchUploader(configuration.cloudwatchEnvironmentName().get()));
```
Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Application.java:30]: can use `java.util.concurrent.ScheduledExecutorService` which can remove Thread handling from CloudWatchUploader.

Comment (philandstuff) [inline src/main/java/uk/gov/indexer/Application.java:30]: :+1: do you want to do this?

Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Application.java:30]: Done!! PR https://github.com/openregister/indexer/pull/41

**Comment by AndrewGorton** [comment]
What  was wrong with the way it was being done?

**Comment by philandstuff** [comment]
@AndrewGorton similar to openregister/mint#65: It's better to use `ScheduledExecutorService` than implement our own scheduling logic.  Among other things, this makes the `CloudwatchProcessHeartbeat` class simpler as it only has one concern now, rather than the two separate concerns of cloudwatch heartbeating and repeated task scheduling.

**Pull request merged**